client/galaxy/scripts/mvc/tool-recommendation.js

```diff
@@ -0,0 +1,237 @@
+                                filteredData.name +
+                                " tool. For further analysis, you could try using the following/recommended tools. The recommended tools are shown in the decreasing order of their scores predicted using machine learning analysis on workflows. A tool with a higher score (closer to 100%) may fit better as the following tool than a tool with a lower score. Please click on one of the following/recommended tools to open its definition. </div>"
+                        );
+                        self.render_tree(filteredData);
+                    } else if (predData["is_deprecated"] === true) {
+                        self.$el.append(
+                            "<div class='warningmessagelarge'>You have used " +
+                                predData.name +
+                                " tool. " +
+                                predData["message"] +
+                                ". </div>"
+                        );
+                    }
+                }
+            }
+        });
+    },
+
+    render_tree: function(predicted_data) {
+        const margin = { top: 20, right: 30, bottom: 20, left: 250 },
+            width = 900 - margin.right - margin.left,
+            height = 300 - margin.top - margin.bottom,
+            duration = 750;
+        const tree = d3.layout.tree().size([height, width]);
+        const diagonal = d3.svg.diagonal().projection(d => {
+            return [d.y, d.x];
+        });
+        const svg = d3
+            .select("#tool-recommendation-view")
+            .append("svg")
+            .attr("width", width + margin.right + margin.left)
+            .attr("height", height + margin.top + margin.bottom)
+            .append("g")
+            .attr("transform", "translate(" + margin.left + "," + margin.top + ")");
+        let i = 0,
+            root = null;
+        function update(source) {
+            // Compute the new tree layout.
+            const nodes = tree.nodes(root).reverse(),
+                links = tree.links(nodes);
+            // Normalize for fixed-depth.
+            nodes.forEach(d => {
+                d.y = d.depth * 180;
+            });
+            // Update the nodes…
+            const node = svg.selectAll("g.node").data(nodes, d => {
+                return d.id || (d.id = ++i);
+            });
+            // Enter any new nodes at the parent's previous position.
+            const nodeEnter = node
+                .enter()
+                .append("g")
+                .attr("class", "node")
+                .attr("transform", d => {
+                    return "translate(" + source.y0 + "," + source.x0 + ")";
+                })
+                .on("click", click);
+            nodeEnter
+                .append("circle")
+                .attr("r", 1e-6)
+                .style("fill", d => {
+                    return d._children ? "lightsteelblue" : "#fff";
```

Comment:
We probably should move these fixed styles and dimensions above, to the top of the file or ideally to css.